### ISSUE TITLE
LDAP Whoops error after clicking Synchronize

### ISSUE THREAD

**Issue by pablom1776** [grokability/snipe-it#8489]
#### Please confirm you have done the following before posting your bug report:

- [ ] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
I do not know how to enable debug mode. I have looked everywhere to see if this was an issued someone else may have posted.
I went to the LDAP in the settings. I ran a Test LDAP Sync and comes back as: Connection to LDAP server established!
I ran a Test LDAP Login: User authentication against LDAP successfully!
When I go to Synchronize in the People tab it takes me to a Whoops error message page, and no users actually sync.
I am wondering if it can be the php. Please provide instructions

**To Reproduce**
1. Go to the People
2. Click on Synchronize
3. See error: Whoops, looks like something went wrong.

**Expected behavior**
Normally I get a list of users that synced.

**Screenshots**
![image](https://user-images.githubusercontent.com/72313638/94997630-73b01280-057a-11eb-9ba9-c3db7837365a.png)

![image](https://user-images.githubusercontent.com/72313638/94997687-c093e900-057a-11eb-8c69-a659f386e6f3.png)

![image](https://user-images.githubusercontent.com/72313638/94997157-6a717680-0577-11eb-9f9b-997e705bdd66.png)

**Server (please complete the following information):**
 - Snipe-IT Version 4.4.2
 - OS: Ubuntu
 - Web Server: apache 2.4.18
 - PHP Version 7.1.18

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Error Messages**
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo.
- Include any additional information you can find in `storage/logs` and your webserver's logs.

**Additional context**
- Is this a fresh install or an upgrade? 
no
- What OS and web server you're running Snipe-IT on
apache 2.4.18 and Ubuntu
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc)
unclear, was already installed before me. It was working fine and all of a sudden stopped.
- Include what you've done so far in the installation, and if you got any error messages along the way.
Not an install, everything was working fine.
- Indicate whether or not you've manually edited any data directly in the database
I have not had to logon to this server for the last 2 years.

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

**Comment by uberbrady** [comment]
It definitely seems strange that the sync **test** works but the actual **sync** doesn't. Especially if nothing else changed.

To enable debug mode, set `APP_DEBUG=true` in your `.env` - then poke around in your `storage/logs/laravel.log` file and see what shows up.

One possibility is if your LDAP server has more than, say, a thousand or so users on it, and your LDAP server doesn't support paging, then we can run into problems - but usually those end up being different problems; a Whoops error is unusual.

Has anything else changed?

**Comment by uberbrady** [comment]
I've *definitely* seen something like this a while ago - where a customer had changed their OU structure (deleting some OU's), and they still had those OU's defined in the Locations. Later versions of Snipe-IT have better error reporting for that.

**Comment by zsirmo** [comment]
I have a same problem. I use(try) MS AD.

Log attached.
[laravel.log](https://github.com/snipe/snipe-it/files/5327627/laravel.log)

**Comment by pablom1776** [comment]
Hello uberbrady,
There have been no changes that I know of. Are you able to send me the exact cmmd I need to run. I do not have a lot of experience with Linux.
Thanks,

**Comment by uberbrady** [comment]
> I have a same problem. I use(try) MS AD.
> 
> Log attached.
> [laravel.log](https://github.com/snipe/snipe-it/files/5327627/laravel.log)

@zsirmo - We fixed that issue in the latest v5 release - https://github.com/snipe/snipe-it/releases/tag/v5.0.0-beta-6-GM - can you give that one a try?

**Comment by uberbrady** [comment]
@pablom1776 it's less of a Snipe-IT issue, and more of a "How is your directory set up?" issue - if you've deleted an OU recently - or renamed one - Snipe-IT will still try to keep looking for it afterwards. We fixed that in a later v4.x release - you can certainly try v4.9.4 or 4.9.5 (though I would recommend 4.9.4 for now).

We had one customer who ended up putting *back* their old OU's and then running a sync, and only after *that* deleting the OU's. That seems excessive to me, though.

v5's sync system is completely rebuilt and shouldn't (emphasis on _shouldn't_) fall prey to the same problems, if you're feeling adventurous. It is not formally released yet, but the above-listed release is what we consider our "Golden Master", hopefully for later this week.

**Comment by zsirmo** [comment]
> > I have a same problem. I use(try) MS AD.
> > Log attached.
> > [laravel.log](https://github.com/snipe/snipe-it/files/5327627/laravel.log)
> 
> @zsirmo - We fixed that issue in the latest v5 release - https://github.com/snipe/snipe-it/releases/tag/v5.0.0-beta-6-GM - can you give that one a try?

Good job, working 👍 

<blockquote><img src="https://repository-images.githubusercontent.com/14346979/164a6580-7d79-11e9-865f-eae87593e11a" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="/snipe/snipe-it/releases/tag/v5.0.0-beta-6-GM">Release v5.0.0-beta-6-GM · snipe/snipe-it</a></strong></div><div>New from v5.0.0-beta-5 to v5.0.0-beta-6-GM
This is the golden master for Snipe-IT v5. Primary in this release are some critical fixes for AD interoperability.
IMPORTANT: This version of Snipe-IT re...</div></blockquote>

**Comment by uberbrady** [comment]
Please let us know if this is still a problem in the latest v5 - https://github.com/snipe/snipe-it/releases/tag/v5.0.4

<blockquote><img src="https://repository-images.githubusercontent.com/14346979/164a6580-7d79-11e9-865f-eae87593e11a" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="/snipe/snipe-it/releases/tag/v5.0.4">Release v5.0.4 · snipe/snipe-it</a></strong></div><div>New in v5.0.4
This is a bug fix release that should handle some of the issues reported over the past few days.
We'll be hitting them back as quickly as you can throw issues at us, so keep them comi...</div></blockquote>

**Comment by pablom1776** [comment]
Thanks for your help everyone. I am good now. I am having an issue with the Backup and the QR code.

**Comment by uberbrady** [comment]
Please open a new issue for your Backup and QR code issues - I suspect they're probably permissions-related.
